codegenerator/cli/npm/envio/src/FetchState.res

```diff
@@ -312,14 +275,17 @@ let updateInternal = (
     normalSelection: fetchState.normalSelection,
     chainId: fetchState.chainId,
     nextPartitionIndex,
-    firstEventBlockNumber,
+    firstEventBlockNumber: switch queue->Utils.Array.last {
+    | Some(item) => Utils.Math.minOptInt(fetchState.firstEventBlockNumber, Some(item.blockNumber))
```

Comment:
Is minOpt necessary here?  Does it take 2 opts? Could you not just pipe queue->last straight in? Just trying to understand the behaviour of the fn